pkg/skaffold/build/list/list.go

```diff
@@ -46,15 +55,26 @@ func Files(workspace string, patterns, excludes []string) ([]string, error) {
 				if err != nil {
 					return err
 				}
-
-				dependencies = append(dependencies, relPath)
+				depSet[relPath] = true
 				return nil
 			}); err != nil {
 				return nil, fmt.Errorf("walking %q: %w", absFrom, err)
 			}
 		}
 	}
 
+	if len(depSet) == 0 {
+		// Adding this conditional in order to be very (overly?) careful with backwards compatibility.
+		// The previous implementation returned nil rather than empty slice in the case of no file dependencies.
```

Comment:
this is *usually* ok in go (the exception is when the resulting slice is directly indexed into, rather than ranged over), but can't hurt to leave this here.